src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let returnString = '';
+    let valueSeparatore = separator;
+
+    if (separator === null) {
+      valueSeparatore = 'null';
+    }
```

Comment:
not sure if you should do it here, I guess if you pass null separator will take it's default value